### PR TITLE
make sure to remove obsolete pending requests

### PR DESCRIPTION
**Pull request by ssalinas**
If statements got switched around a bit back when I did the work for incremental deploys, we used to be removing these. No breakage caused, just annoying extra logging and some extra iterations in the scheduler.

/cc @stevenschlansker @tpetr

### CODE REVIEW

**Comment by tpetr** [comment]
@ssalinas for posterity, could you link to the commit where these things got switched around?

File: SingularityService/src/test/java/com/hubspot/singularity/scheduler/SingularitySchedulerTest.java [hunk 1]
```diff
@@ -2984,4 +2984,18 @@ public void testQueueMultipleOneOffs() {
     Assert.assertEquals(2, taskManager.getPendingTaskIds().size());
   }
 
+  @Test
+  public void testObsoletePendingRequestsRemoved() {
```
Comment (tpetr) [inline SingularityService/src/test/java/com/hubspot/singularity/scheduler/SingularitySchedulerTest.java]: 🏆

**Comment by ssalinas** [comment]
https://github.com/HubSpot/Singularity/commit/7cc9eea559298a4e9cb3907ec7413a2224e5064a#diff-9b41c259c322e21f1971dcdb488de761R224

**Comment by stevenschlansker** [comment]
Any chance this is worth a patch release?  All our masters fill their disks a couple of times a week right now.

**Pull request merged**